fvm/environment/uuids.go

```diff
@@ -39,45 +50,118 @@ func (generator ParseRestrictedUUIDGenerator) GenerateUUID() (uint64, error) {
 
 type uUIDGenerator struct {
 	tracer tracing.TracerSpan
+	log    zerolog.Logger
 	meter  Meter
 
 	txnState state.NestedTransactionPreparer
+
+	blockHeader *flow.Header
+	txnIndex    uint32
+
+	initialized bool
+	partition   byte
+	registerId  flow.RegisterID
+}
+
+func uuidPartition(blockId flow.Identifier, txnIndex uint32) byte {
+	// The deterministically random offset is used to ensure the partitioned
+	// counters are roughly balanced.  Any byte in the sha hash is sufficiently
+	// random/uniform for this purpose.
+	partitionOffset := sha256.Sum256(blockId[:])[0]
+
+	// Partitioning by txnIndex ensures temporally neighboring transactions do
```

Comment:
```suggestion
	// Partitioning by txnIndex ensures temporally neighboring transactions within a block do
```

Reply:
no need to update the comment here.

(if we ever support pipelinging transaction execution across blocks) the randomize partition offset will actually ensure, with high probability, that temporally neighboring transactions across blocks can be executed concurrently